src/AutoRest.CSharp/Mgmt/Generation/ResourceCollectionWriter.cs

```diff
@@ -121,7 +119,8 @@ protected override void WriteCtors()
             using (_writer.WriteMethodDeclaration(parentResourceConstructor))
             {
                 var allPossibleTypes = _resourceCollection.ResourceTypes.SelectMany(p => p.Value).Distinct();
-                _writer.Line($"{ClientDiagnosticsField} = new {typeof(ClientDiagnostics)}(ClientOptions);");
+                string ctorString = ConstructClientDiagnostic(_writer, $"{_resource.Type.Name}.ResourceType.Namespace", DiagnosticOptionsProperty);
```

Comment:
- Please change `ConstructClientDiagnostic` so that it operates with `FormattableString` so that type resolution works properly.
- Use `$"{_resource.Type}` instead of `$"{_resource.Type.Name}`